### PR TITLE
Improve app dev docs

### PR DESCRIPTION
**Pull request by melekes**
Refs #695 

Questions:

1) Can we embed code directly from Github? something like `https://github.com/tendermint/abci/blob/master/example/dummy/persistent_dummy.go#L10-20`
2) I think we need examples in another language(s). Java?
3) persistent-dummy may not be the best example. Other suggestions?

### CODE REVIEW

**Comment by zramsay** [comment]
1. maybe with some javascript but I wouldn't recommend because that file can easily change & keeping track (and updating) of all embedded code blocks would be a PITA. A repo could have self-generating docs (embedding the code at that time) that is then imported by RTD (see last lines of `conf.py`).
2. Ya agree. #701
3. Hmm, we do need to improve upon the dummy/counter/persistent-dummy examples we use. Its mostly a naming issue, no? Maybe we can have a random name for each binary instead?

File: docs/app-development.rst [hunk 1]
```diff
@@ -215,6 +225,22 @@ The block header will be updated (TODO) to include some commitment to
 the results of DeliverTx, be it a bitarray of non-OK transactions, or a
 merkle root of the data returned by the DeliverTx requests, or both.
 
+::
+
+    // tx is either "key=value" or just arbitrary bytes
```
Comment (ebuchman) [inline docs/app-development.rst]: we need to fix this comment because tx can also be `val:pubkey/power`

Comment (melekes) [reply docs/app-development.rst]: fixed in abci:develop

File: docs/app-development.rst [hunk 2]
```diff
@@ -142,6 +142,13 @@ It is unlikely that you will need to implement a client. For details of
 our client, see
 `here <https://github.com/tendermint/abci/tree/master/client>`__.
 
+Most of the examples below are from `dummy application
+<https://github.com/tendermint/abci/blob/master/example/dummy/dummy.go>`__,
+which is a part of the abci repo. `persistent_dummy application
+<https://github.com/tendermint/abci/blob/master/example/dummy/persistent_dummy.go>`__
+application is used to show ``BeginBlock``, ``EndBlock`` and ``InitChain``
```
Comment (ebuchman) [inline docs/app-development.rst]: application twice

File: docs/app-development.rst [hunk 3]
```diff
@@ -228,7 +254,7 @@ Commit, or there will be deadlock. Note also that all remaining
 transactions in the mempool are replayed on the mempool connection
 (CheckTx) following a commit.
 
-The Commit response includes a byte array, which is the deterministic
+The app should respond with a byte array, which is the deterministic
```
Comment (ebuchman) [inline docs/app-development.rst]: "should respond to the Commit request with"

**Pull request merged**